vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewImporter.java

```diff
@@ -984,6 +984,7 @@ private void removeReferences(@NotNull Node node) throws ReferentialIntegrityExc
             if (importMode == ImportMode.REPLACE && !"rep:root".equals(primaryType) && wspFilter.includesProperty(PathUtil.append(node.getPath(), JcrConstants.JCR_PRIMARYTYPE))) {
                 if (!node.getPrimaryNodeType().getName().equals(primaryType)) {
                     vs.ensureCheckedOut();
+                    log.trace("Setting primary node type {} for {}", primaryType, node.getPath());
```

Comment:
Maybe log both original and target primary node type.

Reply:
good point